### DISCUSSION TITLE
Smart schedule for many jobs

### DISCUSSION THREAD

**Discussion by pgrandinetti** [dkron-io/dkron#1225] [Q&A] [answered]
If there are a thousand jobs all scheduled at 00:00, will Dkron start all of them exactly at that time (regardless of how many nodes there are and what their hardware is)?

Related- Are there configuration recommendations for, say, 100s, 1000s and 100,000s of jobs, based on real production systems using Dkron?

Edit: Was missing one zero from "100s"

Thanks!

**Comment by vcastellm** [comment, marked as answer]
Sorry for the late response this slip under my radar.

> If there are a thousand jobs all scheduled at 00:00, will Dkron start all of them exactly at that time (regardless of how many nodes there are and what their hardware is)?

It will try to start them almost at the same time yes, depending on resources it can succeed or fail, 1k is not that much it should handle them with decent hardware.

There's no recommendations or benchmarks published, but according to personal experience and benchmarks it can handle tens of thousands of jobs with medium hardware.
